griffon/services/core_queries.py

```diff
@@ -300,37 +309,129 @@ def execute(self, status=None) -> List[Dict[str, Any]]:
+
+        if self.search_provides:
+            if not self.strict_name_search:
+                params["re_provides_name"] = self.component_name
+            else:
+                params["provides_name"] = self.component_name
+            if self.ns:
+                params["namespace"] = self.ns
+            if not (self.include_inactive_product_streams):
+                params["active_streams"] = "True"
+            params["released_components"] = "True"
+            params["latest_components_by_streams"] = "True"
+            status.update("griffoning: searching latest provided child component(s).")
             latest_components_cnt = self.corgi_session.components.count(**params)
             status.update(f"griffoning: found {latest_components_cnt} latest component(s).")
             latest_components = self.corgi_session.components.retrieve_list_iterator_async(**params)
             with multiprocessing.Pool() as pool:
                 status.update(
-                    f"griffoning: found {latest_components_cnt} latest component(s), retrieving sources & upstreams."  # noqa
+                    f"griffoning: found {latest_components_cnt} latest provides child component(s), retrieving sources & upstreams."  # noqa
                 )
                 for processed_component in pool.map(
                     partial(process_component, self.corgi_session), latest_components
                 ):
                     results.append(processed_component)
             if not self.no_community:
-                status.update("griffoning: searching latest community component(s).")
+                status.update("griffoning: searching latest community provided child component(s).")
                 community_component_cnt = self.community_session.components.count(**params)
                 status.update(
-                    f"griffoning: found {community_component_cnt} latest community component(s)."  # noqa
+                    f"griffoning: found {community_component_cnt} latest community provided child component(s)."  # noqa
                 )
                 latest_community_components = (
                     self.community_session.components.retrieve_list_iterator_async(**params)
                 )
                 with multiprocessing.Pool() as pool:
                     status.update(
-                        f"griffoning: found {community_component_cnt} latest community component(s), retrieving sources & upstreams."  # noqa
+                        f"griffoning: found {community_component_cnt} latest community provided child component(s), retrieving sources & upstreams."  # noqa
+                    )
+                    for processed_component in pool.map(
+                        partial(process_component, self.community_session),
+                        latest_community_components,
+                    ):
+                        results.append(processed_component)
+
+        if self.search_upstreams:
```

Comment:
random observation: for each search section, it seems there is lot of common parts, common params, common query handle, common additional community search, maybe we should aim for some simplification.

Reply:
ya, almost time to rationalise though if it is ok to not do that in this PR